src/main/java/org/opensearch/agent/tools/CreateAnomalyDetectorToolEnhanced.java

```diff
@@ -0,0 +1,1402 @@
+        List<CompletableFuture<Map.Entry<String, Long>>> futures = new ArrayList<>();
+
+        for (String dateField : suggestedDateFields) {
+            CompletableFuture<Map.Entry<String, Long>> future = new CompletableFuture<>();
+            futures.add(future);
+
+            // Count documents with recent data for this date field
+            SearchRequest searchRequest = new SearchRequest(indexName)
+                .source(
+                    new SearchSourceBuilder()
+                        .query(
+                            QueryBuilders
+                                .boolQuery()
+                                .must(QueryBuilders.existsQuery(dateField.trim()))
+                                .must(QueryBuilders.rangeQuery(dateField.trim()).gte(DATE_FIELD_LOOKBACK_PERIOD).lte("now"))
+                        )
+                        .size(0)
+                        .timeout(TimeValue.timeValueSeconds(DATE_FIELD_QUERY_TIMEOUT_SECONDS))
+                );
+
+            client.search(searchRequest, ActionListener.wrap((SearchResponse response) -> {
+                long count = response.getHits().getTotalHits().value();
+                future.complete(Map.entry(dateField.trim(), count));
+            }, e -> {
+                log.warn("Failed to query date field '{}': {}", dateField.trim(), e.getMessage());
+                future.complete(Map.entry(dateField.trim(), 0L));
+            }));
+        }
+
+        // Wait for all queries to complete and select the field with most recent data
+        CompletableFuture
+            .allOf(futures.toArray(new CompletableFuture[0]))
+            .orTimeout(DATE_FIELD_SELECTION_TIMEOUT_SECONDS, java.util.concurrent.TimeUnit.SECONDS)
+            .thenAccept(v -> {
+                String bestDateField = futures
+                    .stream()
+                    .map(CompletableFuture::join)
+                    .max(Comparator.comparing(Map.Entry::getValue))
+                    .map(Map.Entry::getKey)
+                    .orElse(suggestedDateFields[0]); // fallback to first suggestion
+                listener.onResponse(bestDateField);
+            })
+            .exceptionally(e -> {
+                futures.forEach(FutureUtils::cancel);
+                log.error("Date field selection timed out for index '{}', using first suggestion: {}", indexName, e.getMessage());
+                listener.onResponse(suggestedDateFields[0]);
+                return null;
+            });
+    }
+
+    private String createFixPrompt(Map<String, String> originalSuggestions, String validationError) {
+        String currentInterval = originalSuggestions.getOrDefault("interval", "10");
+        String categoryField = originalSuggestions.get(OUTPUT_KEY_CATEGORY_FIELD);
+
+        // Check if this is a sparse data issue with unreasonably high suggested interval
+        boolean isUnreasonableInterval = validationError.contains("interval")
+            && (validationError.contains("240")
```

Comment:
Can we just say the interval is larger than 240, isUnreasonableInterval is true? Suggest API can give a larger number than 1440.